ENIGMAsystem/SHELL/Widget_Systems/None/nowidget_impl.cpp

```diff
@@ -140,12 +140,29 @@ string get_string(string message, string def, string cap="") {
   return (input.empty()) ? def : input;
 }
 
-int get_integer(string message, string def, string cap="") {
+string get_password(string message, string def, string cap="") {
+  //TODO: Add support for hidden input.
   printf("%s\n%s\n", cap.c_str(), message.c_str());
   string input;
   cin >> input;
-  if (input.empty()) input = def;
-  return stoi(input);
+  return (input.empty()) ? def : input;
+}
+
+double get_integer(string message, double def, string cap="") {
+  printf("%s\n%s\n", cap.c_str(), message.c_str());
+  string input;
+  cin >> input;
+  string::size_type sz;
+  return (input.empty()) ? def : stod(input, &sz);
+}
+
+double get_passcode(string message, double def, string cap="") {
+  //TODO: Add support for hidden input.
+  printf("%s\n%s\n", cap.c_str(), message.c_str());
```

Comment:
These lines are identical in function and intended function to `get_password`. In the future, if someone implements `get_password` but does not consider this function, it will stay broken needlessly.

Reply:
I have no idea what you are trying to say here.

Reply:
Allow me to clarify, @JoshDreamland did not see that `get_password` is already implemented with the same code you copied and pasted there. Basically what he wants is for you to extract the common code from both into a helper function and call the helper function. ONLY extract the code that is the same, which is:
`printf("%s\n%s\n", cap.c_str(), message.c_str());`
Extract that into a helper is basically what Josh wants, that way changes to one also apply to the other.

Reply:
Ok, will do!